diff --git a/cmake/externals.cmake b/cmake/externals.cmake
index 9ff2022..83cda6e 100644
--- a/cmake/externals.cmake
+++ b/cmake/externals.cmake
@@ -18,8 +18,8 @@ if(INCLUDE_SANDBOX)
     set(SANDBOX_ARGS -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE} -DCMAKE_INSTALL_PREFIX=${PROJECT_PATH} -DLUA_JIT=off --no-warn-unused-cli)
     externalproject_add(
         ${SANDBOX_PACKAGE}
-        GIT_REPOSITORY https://github.com/mozilla-services/lua_sandbox.git
-        GIT_TAG 82ae00d85f3111ee77bf15dbcb97264f177a66ec
+        GIT_REPOSITORY https://github.com/jedisct1/lua_sandbox.git
+        GIT_TAG rfc5424-sd
         CMAKE_ARGS ${SANDBOX_ARGS}
         INSTALL_DIR ${PROJECT_PATH}
     )
diff --git a/sandbox/lua/decoders/rsyslog.lua b/sandbox/lua/decoders/rsyslog.lua
index d6a36ff..b3256f8 100644
--- a/sandbox/lua/decoders/rsyslog.lua
+++ b/sandbox/lua/decoders/rsyslog.lua
@@ -105,6 +105,12 @@ function process_message ()
     msg.Payload = fields.msg
     fields.msg = nil
 
+    local sd = fields["structured-data"]
+    if sd and type(sd) == "table" then
+        for k, v in pairs(sd) do fields[k] = v end
+        fields["structured-data"] = nil
+    end
+
     msg.Fields = fields
     if not pcall(inject_message, msg) then return -1 end
     return 0
