diff --git a/cmake/externals.cmake b/cmake/externals.cmake
index 9ff2022..c7fd349 100644
--- a/cmake/externals.cmake
+++ b/cmake/externals.cmake
@@ -19,7 +19,7 @@ if(INCLUDE_SANDBOX)
     externalproject_add(
         ${SANDBOX_PACKAGE}
         GIT_REPOSITORY https://github.com/mozilla-services/lua_sandbox.git
-        GIT_TAG 82ae00d85f3111ee77bf15dbcb97264f177a66ec
+        GIT_TAG 8602e0a6fb098f083d8a80947dce0320fe70b0a3
         CMAKE_ARGS ${SANDBOX_ARGS}
         INSTALL_DIR ${PROJECT_PATH}
     )
diff --git a/sandbox/lua/decoders/rsyslog.lua b/sandbox/lua/decoders/rsyslog.lua
index d6a36ff..b3256f8 100644
--- a/sandbox/lua/decoders/rsyslog.lua
+++ b/sandbox/lua/decoders/rsyslog.lua
@@ -105,6 +105,12 @@ function process_message ()
     msg.Payload = fields.msg
     fields.msg = nil
 
+    local sd = fields["structured-data"]
+    if sd and type(sd) == "table" then
+        for k, v in pairs(sd) do fields[k] = v end
+        fields["structured-data"] = nil
+    end
+
     msg.Fields = fields
     if not pcall(inject_message, msg) then return -1 end
     return 0
